external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDataToCatalyst.scala

```diff
@@ -53,7 +54,9 @@ case class AvroDataToCatalyst(
 
   @transient private lazy val avroSchema = new Schema.Parser().parse(jsonFormatSchema)
 
-  @transient private lazy val reader = new GenericDatumReader[Any](avroSchema)
+  @transient private lazy val reader = writerJsonFormatSchema
```

Comment:
It's `writerJsonFormatSchema` but why we need to set in reader?

Reply:
It's the whole point of the PR, it's explained in the description and in the Jira ticket

Reply:
I am saying it doesn't look clear unless you read the whole PR description and JIRA.